sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/EmbeddingItem.java

```diff
@@ -37,7 +63,10 @@ public List<Float> getEmbedding() {
      * @return the embedding base64 encoded string.
      */
     public String getEmbeddingAsString() {
-        return embeddingBase64;
+        if (embeddingBase64 != null) {
+            return embeddingBase64;
+        }
+        return convertFloatListToBase64(embedding.stream().map(Double::floatValue).collect(Collectors.toList()));
```

Comment:
Same here - we should convert only once.

```suggestion
        embeddingBase64 = convertFloatListToBase64(embedding.stream().map(Double::floatValue).collect(Collectors.toList()));
        return embeddingBase64;
```